src/app/organizations/manage/events.component.html

```diff
@@ -12,17 +12,21 @@ <h1>{{'eventLogs' | i18n}}</h1>
                 placeholder="{{'endDate' | i18n}}" [(ngModel)]="end" placeholder="YYYY-MM-DDTHH:MM"
                 (change)="dirtyDates = true">
         </div>
-        <button #refreshBtn [appApiAction]="refreshPromise" type="button" class="btn btn-sm btn-outline-primary ml-3"
-            (click)="loadEvents(true)" [disabled]="loaded && refreshBtn.loading">
-            <i class="fa fa-refresh fa-fw" aria-hidden="true" [ngClass]="{'fa-spin': loaded && refreshBtn.loading}"></i>
-            {{'refresh' | i18n}}
-        </button>
-        <button #exportBtn [appApiAction]="exportPromise" type="button"
-            class="btn btn-sm btn-outline-primary btn-submit manual ml-3" [ngClass]="{loading:exportBtn.loading}"
-            (click)="exportEvents()" [disabled]="loaded && exportBtn.loading || dirtyDates">
-            <i class="fa fa-spinner fa-spin" aria-hidden="true"></i>
-            <span>{{'export' | i18n}}</span>
-        </button>
+        <form #refreshForm [appApiAction]="refreshPromise" class="d-inline">
+            <button type="button" class="btn btn-sm btn-outline-primary ml-3" (click)="loadEvents(true)"
+                [disabled]="loaded && refreshForm.loading">
+                <i class="fa fa-refresh fa-fw" aria-hidden="true" [ngClass]="{'fa-spin': loaded && refreshForm.loading}"></i>
+                {{'refresh' | i18n}}
+            </button>
+        </form>
+        <form #exportForm [appApiAction]="exportForm" class="d-inline">
+            <button type="button" class="btn btn-sm btn-outline-primary btn-submit manual ml-3"
+                [ngClass]="{loading:refreshForm.loading}" (click)="exportEvents()"
```

Comment:
This has changed from `{loading:exportBtn.loading}` to `{loading:refreshForm.loading}` - was that intentional?

Is this `[ngClass]` directive still required now that this is wrapped in `<form>` elements? My understanding is that the `loading` property now exists on the `<form>` element - is there anything still testing `loading` on the button?

Reply:
Nah, I wish I could get rid of it, but you still need it because we're disabling the button for more than just loading reasons -- It's also disabled when dates entered in the form are dirty.

The `manual` class overrides the `btn-submit` class' default behavior of changing the button to a loading circle whenever it is disabled.  The `[ngclass]` directive adds on the loading class (which changes the button to a spinner) when the form is in a loading state.

Here's the SASS for it, which can explain better than I can: https://github.com/bitwarden/web/blob/master/src/scss/styles.scss#L345-L368

Reply:
Alrighty! As you were